webapi/src/main/scala/org/knora/webapi/messages/OntologyConstants.scala

```diff
@@ -655,13 +655,28 @@ object OntologyConstants {
         val KnoraApiPrefix: String = KnoraApiOntologyLabel + ":"
 
         /**
-         * Returns `true` if the specified IRI is `knora-api:Resource` in Knora API v2, in the simple
-         * or complex schema.
+         * The IRIs representing `knora-api:Resource` in Knora API v2, in the simple and complex schemas.
          */
-        def isKnoraApiV2Resource(iri: SmartIri): Boolean = {
-            val iriStr = iri.toString
-            iriStr == OntologyConstants.KnoraApiV2Simple.Resource || iriStr == OntologyConstants.KnoraApiV2Complex.Resource
-        }
+        lazy val KnoraApiV2ResourceIris: Set[IRI] = Set(
```

Comment:
@benjamingeer question: why are using `lazy val` here in the following three cases? The initialization does not need any evaluation of an expression, it is just a set. In this case, it would not make any difference if the initialization happens immediately or is postponed to the first time the value is accessed.

Reply:
Great question! The problem is that `KnoraApiV2ResourceIris` is a static value (it's defined in an `object` rather than in a `class`), which refers to static values in another `object`. The order of initialisation of static values in the JVM depends on which `object` is loaded first by the JVM's class loader at run time. Without the `lazy`, if `object KnoraApi` is initialised before `object KnoraApiV2Simple`, `KnoraApiV2ResourceIris` will just contain `null` values. It took me a little while to figure this out yesterday. A test was failing, and it turned out that it was because `KnoraApiV2ResourceIris` did not contain `OntologyConstants.KnoraApiV2Complex.Resource`, because it hadn't been initialised properly.

In the earlier code, there was a method, which didn't access those contents until it was called, so it was fine.